src/WebJobs.Script.WebHost/FunctionsSyncService.cs

```diff
@@ -21,17 +21,19 @@ public class FunctionsSyncService : IHostedService, IDisposable
         private readonly IScriptHostManager _scriptHostManager;
         private readonly IPrimaryHostStateProvider _primaryHostStateProvider;
         private readonly IFunctionsSyncManager _functionsSyncManager;
+        private readonly IEnvironment _environment;
         private Timer _syncTimer;
         private bool _disposed = false;
 
-        public FunctionsSyncService(ILoggerFactory loggerFactory, IScriptHostManager scriptHostManager, IPrimaryHostStateProvider primaryHostStateProvider, IFunctionsSyncManager functionsSyncManager)
+        public FunctionsSyncService(ILoggerFactory loggerFactory, IScriptHostManager scriptHostManager, IPrimaryHostStateProvider primaryHostStateProvider, IFunctionsSyncManager functionsSyncManager, IEnvironment environment)
```

Comment:
Thinking we don't need to modify this ctor at all. Instead, we change [here ](https://github.com/Azure/azure-functions-host/blob/ce4fa5fe4b468123ed3419e106719ac614d10edb/src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs#L109)to not even register the service in validation mode. I.e. just modify FunctionsSyncManager.IsSyncTriggersEnvironment to do your new check.

Reply:
Ah, you may be right - I was thinking that these validation pods were special in some way and that this state would be known before specialization.